outputs.tf

```
@@ -53,6 +53,22 @@ output "appsync_function_function_id" {
   value       = { for k, v in aws_appsync_function.this : k => v.function_id }
 }
 
+# Domain
+output "appsync_domain_id" {
+  description = "The Appsync Domain Name."
+  value       = var.domain_name_association_enabled ? aws_appsync_domain_name.this[0].id : null
```

```suggestion
  value       = try(aws_appsync_domain_name.this[0].id, null)
```
This is better. Please apply a similar fix on other outputs, too. Including what was there before (e.g. `appsync_graphql_api_id` and others).

In some cases, a ternary operation can be broken when resource creation failed but some argument is already being requested. `try` is more optimistic in this case.

---

yeah, I don't know what I was thinking. you're absolutely right. Thanks for the feedback :)